pkg/app/pipedv1/plugin/kubernetes/server.go

```diff
@@ -16,31 +16,38 @@ package main
 
 import (
 	"context"
+	"net"
+	"strconv"
 	"time"
 
 	"github.com/pipe-cd/pipecd/pkg/app/pipedv1/plugin/kubernetes/deployment"
+	"github.com/pipe-cd/pipecd/pkg/app/pipedv1/plugin/toolregistry"
 	"github.com/pipe-cd/pipecd/pkg/cli"
+	"github.com/pipe-cd/pipecd/pkg/plugin/logpersister"
+	"github.com/pipe-cd/pipecd/pkg/plugin/pipedapi"
 	"github.com/pipe-cd/pipecd/pkg/rpc"
 	"github.com/spf13/cobra"
 	"go.uber.org/zap"
 	"golang.org/x/sync/errgroup"
 )
 
 type server struct {
-	apiPort     int
-	gracePeriod time.Duration
-	tls         bool
-	certFile    string
-	keyFile     string
+	apiPort                int
+	pipedPluginServicePort int
+	gracePeriod            time.Duration
+	tls                    bool
+	certFile               string
+	keyFile                string
 
 	enableGRPCReflection bool
 }
 
 // NewServerCommand creates a new cobra command for executing api server.
 func NewServerCommand() *cobra.Command {
 	s := &server{
-		apiPort:     10000,
-		gracePeriod: 30 * time.Second,
+		apiPort:                10000,
+		pipedPluginServicePort: 9087,
```

Comment:
[ASK] Where does `9087` come from?

Do other plugins need to avoid using `9087`?

Reply:
Sorry for not commenting about this.
This comes from the default of pipedv1's plugin service port.
https://github.com/pipe-cd/pipecd/blob/4986dd5158d424bc841c19713094e611b1235479/pkg/app/pipedv1/cmd/piped/piped.go#L103

Reply:
I got it, thank you.
I think it'd be better to avoid hard coding (e.g., by using const) because all plugins need to specify 9087 directly.

If it's not easy now, let's do it in another PR later.

Reply:
I came to one idea.
The piped will always specify the port.
Then, making the default `-1` and the plugin raises the error if the piped doesn't specify the port.
WDYT?

Reply:
It seems good!

You mean a piped passes the piped's port to each plugin, right?  If not passed, an error will occur by `-1`

Reply:
Yes 👍🏻 
I realized the way to pass this port must be the same across all the plugins.
We have to discuss how to pass it (e.g., as a command option or an environment variable, and these names, too).

Reply:
@t-kikuc
I changed the default to -1 and added validation.
In this PR, I want to merge it with this implementation.

https://github.com/pipe-cd/pipecd/pull/5387/commits/81bef1814ee9a775d9a92d34c4c7c8b89e9012f6

Reply:
@Warashi Thank you!